packages/react-inbox/src/hooks/use-inbox.ts

```diff
@@ -4,18 +4,19 @@ import useMessages from "~/hooks/use-messages";
 
 export default () => {
   const { fetch: fetchMessages } = useMessages();
-  const {
-    dispatch, inbox, transport,
-  } = useCourier();
+  const { dispatch, inbox, transport } = useCourier();
 
   const [_, trackEvent] = useTrackEvent();
 
-  const newMessage = useCallback((payload) => {
-    dispatch({
-      type: "inbox/NEW_MESSAGE",
-      payload,
-    });
-  }, [dispatch]);
+  const newMessage = useCallback(
+    (payload) => {
+      dispatch({
+        type: "inbox/NEW_MESSAGE",
+        payload,
+      });
+    },
+    [dispatch]
```

Comment:
do we need useCallback here?

Reply:
because everytime we called this function, it was creating a new function and causing extra rerenders. 

also, eslint told me to